### PR TITLE
Mateo.manolio/module 4

### PR DESCRIPTION
**Pull request by MateoManolio**
Estructura de archivos basada en **clean architecture**
image

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -1,36 +1,35 @@
 import 'package:flutter/material.dart';
-import 'package:movie_card/pages/home_page.dart';
-import 'pages/movie_details.dart';
-import 'shared/movie.dart';
+import 'package:movie_card/src/presentation/pages/home_page.dart';
```
Comment (matiasilveiro-gl) [inline lib/main.dart]: Relative import, to keep consistency

File: lib/main.dart [hunk 2]
```diff
@@ -1,36 +1,35 @@
...
 }
 
 class MyApp extends StatelessWidget {
-  const MyApp({super.key});
+  const MyApp({
+    super.key,
+  });
 
-  static const routeName = '/';
+  static const String routeName = '/';
```
Comment (matiasilveiro-gl) [inline lib/main.dart]: This name is not clear enough, it's too generic... What does it represent? Should it be here?
Why not calling it `initialRoute` and have it inside `AppRoutes`?

File: lib/src/config/route/app_routes.dart [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+abstract class AppRoutes {
+  static Map<String, WidgetBuilder> get routes {
+    return <String, WidgetBuilder>{
+      MovieDetails.routeName: (BuildContext context) =>
+          MovieDetails(
+            movie: MovieModel.mockMovie(),
+          ),
+    };
+  }
+}
```
Comment (matiasilveiro-gl) [inline lib/src/config/route/app_routes.dart]: EOF

File: lib/src/config/route/app_routes.dart [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/material.dart';
+
+import '../../data/models/movie_model.dart';
+import '../../presentation/pages/movie_details.dart';
+
+abstract class AppRoutes {
+  static Map<String, WidgetBuilder> get routes {
```
Comment (matiasilveiro-gl) [inline lib/src/config/route/app_routes.dart:10]: I'd put the "HomePage" route here as well.

File: lib/src/config/theme/app_themes.dart [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import 'package:flutter/material.dart';
+
+abstract class AppTheme {
+  static ThemeData get dark {
+    return ThemeData(
+      colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
+      useMaterial3: true,
+    );
+  }
+}
```
Comment (matiasilveiro-gl) [inline lib/src/config/theme/app_themes.dart]: EOF. Configure your IDE to automatically add a line feed at the end of the file

File: lib/src/data/models/cast_model.dart [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  );
+
+  static List<CastModel> fromListOfJson(List<dynamic> list) {
+    List<CastModel> casts = List.generate(
+      list.length,
+          (index) => CastModel.fromJson(list[index]),
+    );
+    return casts;
+  }
+}
```
Comment (matiasilveiro-gl) [inline lib/src/data/models/cast_model.dart]: EOF

File: lib/src/data/models/cast_model.dart [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import '../../domain/entity/cast.dart';
+
+class CastModel extends Cast {
+  CastModel({required super.id, required super.profilePath,});
```
Comment (matiasilveiro-gl) [inline lib/src/data/models/cast_model.dart]: Formatting

File: lib/main.dart (hunk 1 above)
Comment (MateoManolio) [reply lib/main.dart]: Fixed

File: lib/main.dart (hunk 2 above)
Comment (MateoManolio) [reply lib/main.dart]: No, it should't, so i erase it 😄

File: lib/src/config/route/app_routes.dart (hunk 3 above)
Comment (MateoManolio) [reply lib/src/config/route/app_routes.dart]: Fixed

File: lib/src/config/route/app_routes.dart (hunk 4 above)
Comment (MateoManolio) [reply lib/src/config/route/app_routes.dart:10]: Done

File: lib/src/config/theme/app_themes.dart (hunk 5 above)
Comment (MateoManolio) [reply lib/src/config/theme/app_themes.dart]: Fixed

File: lib/src/data/models/cast_model.dart (hunk 6 above)
Comment (MateoManolio) [reply lib/src/data/models/cast_model.dart]: Fixed

File: lib/src/data/models/cast_model.dart (hunk 7 above)
Comment (MateoManolio) [reply lib/src/data/models/cast_model.dart]: Fixed

**Review by MOLLL009 — APPROVED** [review]

**Review by Kevin-Guillen — APPROVED** [review]

**Review by matiasilveiro-gl — APPROVED** [review]

**Review by Kevin-Guillen — APPROVED** [review]

**Review by MOLLL009 — APPROVED** [review]

**Pull request merged**